python/runtime/common/py_NoiseModel.cpp

```diff
@@ -80,46 +113,65 @@ void bindNoiseModel(py::module &mod) {
              }
              return kraus_channel(kops);
            }),
-           "Create a KrausChannel given a list of KrausOperators.")
+           "Create a :class:`KrausChannel` given a list of "
+           ":class:`KrausOperator`'s.")
       .def(
           "__getitem__",
-          [](kraus_channel &c, std::size_t idx) { return c[idx]; },
-          "Return the KrausOperator at the given index in this KrausChannel.")
+          [](kraus_channel &self, std::size_t idx) { return self[idx]; },
+          py::arg("index"),
+          "Return the :class:`KrausOperator` at the given index in this "
+          ":class:`KrausChannel`.")
       .def("append", &kraus_channel::push_back,
-           "Add a KrausOperator to this KrausChannel.");
-
-  py::class_<noise_model>(
-      mod, "NoiseModel",
-      "The cudaq NoiseModel defines a set of KrausChannels applied to "
-      "specific qubits after the invocation specified quantum operations.")
-      .def(py::init<>(), "Construct an empty noise model.")
-      .def(
-          "add_channel",
-          [](noise_model &n, std::string &opName,
-             std::vector<std::size_t> &qubits,
-             kraus_channel &c) { n.add_channel(opName, qubits, c); },
-          "Add the given KrausChannel to be applied after invocation of the "
-          "specified quantum operation.")
-      .def(
-          "get_channels",
-          [](noise_model n, const std::string &op,
-             std::vector<std::size_t> qbits) {
-            return n.get_channels(op, qbits);
-          },
-          "Return the KrausChannels that make up this noise model.");
-
-  py::class_<depolarization_channel, kraus_channel>(mod,
-                                                    "DepolarizationChannel")
-      .def(py::init<double>());
+           "Add a KrausOperator to this :class:`KrausChannel`.");
+
+  py::class_<depolarization_channel, kraus_channel>(
+      mod, "DepolarizationChannel",
+      "Models the decoherence of the qubit state and phase into a mixture "
+      "of the computational basis states, `|0>` and `|1>`. Its constructor "
```

Comment:
Maybe we should mention here that it is modeled as uniform Pauli errors (p/3 for X, Y, Z).